components/lib/toast/Toast.js

```diff
@@ -19,13 +19,11 @@ export const Toast = React.memo(
 
         const show = (messageInfo) => {
             if (messageInfo) {
-                const messages = assignIdentifiers(messageInfo, true);
-
-                setMessagesState(messages);
+                setMessagesState((previousMessagesState) => assignIdentifiers(previousMessagesState, messageInfo, true));
             }
```

Comment:
Can you apply similar fixes to the Messages component as it works the exact same.

Reply:
Done, but there is a lot of duplication in these files. Some refactoring here would help.